### PR TITLE
Add commands for building xchammer.app with bazel

### PR DESCRIPTION
**Pull request by rahul-malik**
- Update CI tests to use xchammer.app
- Fix lookups to resources folders

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -136,28 +147,28 @@ SAMPLE ?= UrlGet
 # Run a debug build of XCHammer against a sample
 # Development hack: don't actually install, just symlink the debug build
 # See README for usage in a normal project
-run: build
-	$(ROOT_DIR)/.build/debug/$(PRODUCT) generate \
+run: build_bazel
```
Comment (jerrymarino) [inline Makefile]: Don't add this to `make run` until we have incremental builds working

File: Makefile [hunk 2]
```diff
@@ -187,8 +198,10 @@ goldmaster:
 		ditto sample/$$S/$$S.xcodeproj/xcshareddata/xcschemes $$MASTER/xcshareddata/xcschemes; \
 	done
 
-run_swift: build
-	$(ROOT_DIR)/.build/debug/$(PRODUCT) generate \
+
+
+run_swift: build_bazel
```
Comment (jerrymarino) [inline Makefile]: This can be deleted altogether, just use `SAMPLE=UrlGet make run`

Comment (jerrymarino) [reply Makefile]: `make run SAMPLE=Tailor`

File: Sources/XCHammer/Generator.swift [hunk 3]
```diff
@@ -757,7 +757,7 @@ enum Generator {
     }
 
     private static func getTulsiAspectRepo() -> String {
-        return Bundle.main.bundlePath
+        return getAssetBase()
```
Comment (jerrymarino) [inline Sources/XCHammer/Generator.swift]: Does this work with the `SPM` build still? I haven't tested it out yet, and we should make sure it's supported before merging. I thought we needed to look in `tulsi-aspects`..

Comment (rahul-malik) [reply Sources/XCHammer/Generator.swift]: Yep it still works, manually verified and CI has verified it as well

File: Makefile [hunk 4]
```diff
@@ -1,20 +1,32 @@
...
+	PRODUCT=XCHammer
+	XCHAMMER_BIN_BASE=$(PWD)/.build/debug
+endif
+
+XCHAMMER_BIN := $(XCHAMMER_BIN_BASE)/$(PRODUCT)
+
+PREFIX := /usr/local
+
 aspects:
+ifeq ($(BAZEL_BUILD),false)
```
Comment (jerrymarino) [inline Makefile]: `ifneq ($(BAZEL_BUILD),true)` - this won't be set to `false` by default

**Pull request merged**